### PR TITLE
French tutorial

### PR DESCRIPTION
**Pull request by jemsab**
Translation of English tutorial into French, PR related to issue Out of date French documentation #24

### CODE REVIEW

**Comment by jemsab** [comment]
Hi,
I've translated the English tutorial_en.catala_en into French tutorial_fr.catala_fr.
For the moment the French tutorial is in the same directory as the original English one.

Do you want me to move it to a separate folder? 
If not, I guess I should modify the tutorial makefile to handle multiple languages. What do you think?

**Comment by denismerigoux** [comment]
Thanks @jemsab ! This translation looks amazing. Indeed the Makefile structure I did does not handle multi-language examples. It would be hard and cumbersome to modify the Makefile to support it so for me the simplest solution is to move the English tutorial into an `tutorial_en` directory and the French tutorial into a `tutoriel_fr` directory.

**Comment by jemsab** [comment]
Thanks! I have split the two tutorials but I'm having some trouble compiling the French version.  Are some symbols not implemented in the French lexer ?

I get this error :

    λ ~/Code/catala/examples/tutoriel_fr/ french-doc-translation* make tutorial_fr.tex    [ERROR] Parsing error after token "*": what comes after is unknown
    [ERROR] 
    [ERROR]    --> tutorial_fr.catala_fr
    [ERROR]     | 
    [ERROR] 134 |     personne.revenu *$ pourcentage_fixe
    [ERROR]     |                     ^
    make: *** [tutorial_fr.tex] Error 255

**Comment by denismerigoux** [comment]
Ha ha, try `*€` instead, I translated things in the mexer too :)

**Comment by jemsab** [comment]
I did wonder about that. I imagine all monetary units and operations should be expressed with €?

**Comment by denismerigoux** [comment]
Yep, and the French lexer also follows the French convention for money literals like `250 000.63 €`

**Comment by denismerigoux** [comment]
Here are the relevant bits from the French lexer:

https://github.com/CatalaLang/catala/blob/33f62fefc01b82cd8f8281e9f740b7b8306c11bf/src/catala/catala_surface/lexer_fr.ml#L356-L358

https://github.com/CatalaLang/catala/blob/33f62fefc01b82cd8f8281e9f740b7b8306c11bf/src/catala/catala_surface/lexer_fr.ml#L277-L292

**Comment by denismerigoux** [comment]
The CI error (https://github.com/CatalaLang/catala/runs/1604743253#step:8:1550) stems from the fact that the root Makefile has to be updated whenever examples directories change names: 

https://github.com/jemsab/catala/blob/f0cfe89d2b74ea7e76cdf6ac70c6a04751a10508/Makefile#L126-L128

File: examples/tutoriel_fr/tutoriel_fr.catala_fr [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+Si vous écriviez un programme Catala pour une loi française, vous devrez utiliser
+l'extension ".catala_fr".
+
+Vous pouvez écrire n'importe quel texte simple en Catala, cela sera affiché
+sans modification dans une sortie PDF ou HTML. Vous pouvez découper votre texte
+en de courtes lignes, cela apparaîtera comme un seul paragraphe dans la sortie.
+Si vous voulez créer un nouveau paragraphe laisser une ligne vierge dans la source.
+
+Catala vous permet de déclarer des entêtes de section ou de sous-section, 
+comme c'est le cas ici, acec la répétition deux fois de l'arobase. Vous pouvez
```
Comment (denismerigoux) [inline examples/tutoriel_fr/tutoriel_fr.catala_fr:25]: Typo

File: examples/tutoriel_fr/tutoriel_fr.catala_fr [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+@Article 3@ Si l'individu a à sa charge deux ou plus enfants alors
+le pourcentage fixe mentionné à l'article 1 vaut 15 %.
+/*
+# Comment redéfinir pourcentage_fixe ?
+*/
+
+Cet article donne en réalité une autre définition pour le pourcentage fixe, 
+préalablement défini à l'article 2. Toutefois, si l'article 3 définit le pourcentage
+de manière conditionnelle pour la personne ayant plus de deux enfants. 
+Catala permet de redéfinir précisément uen variable sous une condition :
```
Comment (denismerigoux) [inline examples/tutoriel_fr/tutoriel_fr.catala_fr:187]: Typo

File: examples/tutoriel_fr/tutoriel_fr.catala_fr [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+  définition pourcentage_fixe sous condition
+    personne.nombre_enfants >= 2
+  conséquence égal à 15 %
+  # Ecrire 15% est juste une abbréviation pour 0.15
+*/
+
+Quand le programme Catala va s'exécuter, la définition qui est juste sera choisi
+dynamiquement en vérifiant quelle condition est vraie. Une source législative
+rédigée correctement doit toujours garantir qu'au maximum une seule condition 
+soit vrai à tout moment. Toutefois si ce n'est pas le cas, Catala vous permettra
```
Comment (denismerigoux) [inline examples/tutoriel_fr/tutoriel_fr.catala_fr:200]: vrai**e**

File: examples/tutoriel_fr/tutoriel_fr.catala_fr [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+La loi ne le précise pas; nos articles sont clairement mal rédigés.
+Mais Catala vous aide à trouver ce genre d'erreur par de simples tests ou
+même la vérification formelle. Commençons par les tests.
+
+@@Tester les programmes Catala@@+
+
+Tester les programmes Catala peut se faire directement en Catala. En effet,
+écrire des cas de tests pour chaque champ d'application Catala que vous 
+définissez est une bonne pratique appelée "tests unitaires" dans la 
+communauté du génie logicielle. Un cas de test est défini dans un autre
```
Comment (denismerigoux) [inline examples/tutoriel_fr/tutoriel_fr.catala_fr:294]: *logiciel* au masculin

**Review by denismerigoux — APPROVED** [review]
I proof-read the translation, perfect apart from some minor typos!

**Comment by denismerigoux** [comment]
I'll merge it now, thanks for the contribution and Merry Christmas!

**Comment by jemsab** [comment]
Glad to help, Merry Christmas!

**Pull request merged**